kymatio/scattering2d/frontend/torch_f.py

```diff
@@ -0,0 +1,178 @@
+        M_padded, N_padded : int
+             spatial support of the padded input
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8
+
+        pre_pad is particularly useful when doing crops of a bigger
+         image because the padding is then extremely accurate. Defaults
+         to False.
+
+        """
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False):
+        super(Scattering2D, self).__init__(J, shape, max_order = max_order)
+        self.L, self.J, self.max_order = L, J, max_order
+        self.pre_pad = pre_pad
+        self.shape = shape
+        if 2 ** J > shape[0] or 2 ** J > shape[1]:
+            raise (RuntimeError('The smallest dimension should be larger than 2^J'))
+        self.build()
+
+    def build(self):
+        self.M, self.N = self.shape
+        self.modulus = Modulus()
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = Pad([(self.N_padded - self.N) // 2, (self.N_padded - self.N + 1) // 2, (self.M_padded - self.M) // 2,
+                       (self.M_padded - self.M + 1) // 2], [self.N, self.M], pre_pad=self.pre_pad)
+        self.subsample_fourier = SubsampleFourier()
+        self.filters = self.create_and_register_filters()
+
+    def create_and_register_filters(self):
+        # Create the filters
+        filters = filter_bank(self.M_padded, self.N_padded, self.J, self.L)
+        n = 0
+        self.phi, self.psi = filters['phi'], filters['psi']
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c])
+                self.register_buffer('tensor' + str(n), self.phi[c])
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = torch.from_numpy(v)
+                    self.register_buffer('tensor' + str(n), self.psi[j][k])
+                    n += 1
+
+    def scattering(self, input):
+        # each time scattering is run, one needs to make sure self.psi and self.phi point to
+        # the correct buffers
+        n=0
+        buffer_dict = dict(self.named_buffers())
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] =  buffer_dict['tensor' + str(n)]
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = buffer_dict['tensor' + str(n)]
+                    n += 1
+
+        return scattering2d(input, self.J, self.L, self.subsample_fourier, self.pad, self.modulus, fft, cdgmm, unpad, self.phi, self.psi, self.max_order, self.M_padded, self.N_padded)
```

Comment:
Why not just pass `self` and `backend`? Long argument lists like this are rarely a good idea.

Reply:
I quite disagree here! It is the simplest way to make the function `scattering2d` self-suffisant. Otherwise, for other frontends/backends, we might be missing some modules. Also, frontends which do not have a "class" structure might suffer from this.(e.g., tf)

Reply:
Ok sure. But can we at least package things into a dict or something instead of a long argument list?

Reply:
Sure..